### DISCUSSION TITLE
abort() function does not log exceptions in laravel.log

### DISCUSSION THREAD

**Discussion by akshit-arora** [laravel/framework#49033] [Q&A] [answered]
Hi all,

I am not able to see the logs if I use `abort()` function. The application logs the exceptions if I throw any exception but not when using `abort()` function

Can someone please help me or share any references which mentions that abort won't log the error in laravel.log.

Thanks in advance!

**Comment by antimech** [comment, marked as answer]
`abort()` throws `HttpResponseException` [which is on the $internalDontReport list in the Illuminate\Foundation\Exceptions\Handler::class](https://github.com/laravel/framework/blob/e72327a0ad074a2d9d56742e6c7ff9eab1042528/src/Illuminate/Foundation/Exceptions/Handler.php#L109).

**Reply by akshit-arora** [reply]
Thank you. Is there any way by which we can make it reportable? May be by passing a flag or setting some config?

I can see a function that can help us in that `stopIgnoring()` but can't figure a way to use that in this case.
[Here's the reference to the function](https://github.com/laravel/framework/blob/e72327a0ad074a2d9d56742e6c7ff9eab1042528/src/Illuminate/Foundation/Exceptions/Handler.php#L385)

**Reply by antimech** [reply]
This way you will get a whole bunch of unneeded errors into your logs, like 404. My guess is that you looking for creating your own exception:
```Shell
php artisan make:exception InvalidOrderException
```

**Reply by akshit-arora** [reply]
Hmm.. makes sense. I was using that when fetching information from service classes in all my controllers.

```php
try {
    $response = $this->myServiceClass->someFunctionThatCrashes();
} catch (Exception $exception) {
    abort(Response::HTTP_INTERNAL_SERVER_ERROR, $exception->getMessage());
}
```
But now what's happening is that I can't figure out why the application crashed as there are no log entries for that. I got the workaround on Twitter by @davidbirkin for this to use `report($exception)` function before `abort()` to get that error logged. But was looking if I can have any alternative for this.

**Reply by antimech** [reply]
```php
try {
    throw new InvalidOrderException();
} catch (InvalidOrderException $e) {
    report($e);
}

echo('It did not stop but logged the error.');
```

**Reply by antimech** [reply]
Just in case you ever need to [stop ignoring some exception](https://laravel.com/docs/10.x/errors#ignoring-exceptions-by-type). Put it in your exception handler (`App\Exceptions\Handler`):
```php
use Symfony\Component\HttpKernel\Exception\HttpException;
 
/**
 * Register the exception handling callbacks for the application.
 */
public function register(): void
{
    $this->stopIgnoring(HttpException::class);
 
    // ...
}
```

How else can I help you?

**Reply by davidbirkin** [reply]
@antimech Thanks for this additional information, helps me too! :)
